package NYql;

message TFileStorageConfig {
    optional string Path = 1; // Path to file storage. An auto-cleaned temp directory is used for empty value
    optional uint32 MaxFiles = 2 [default = 1000]; // Maximum number of files in the storage
    optional uint32 MaxSizeMb = 3 [default = 100]; // Maximum total size of all files in the storage
    optional uint32 Threads = 5 [default = 1]; // Number of download threads for async downloading
    optional uint32 RetryCount = 6 [default = 0]; // Number of additional attempts to download file
    optional uint32 RetryDelayMs = 7 [default = 1000]; // Delay in ms between attempts to download file

    map<string, string> DownloaderConfig = 14;
}
